### PR TITLE
validation: Validate we specify ids for all subnets

### PR DESCRIPTION
**Pull request by justinsb**
Move our validation to the apimachinery style.  And then add a
validation that we specify IDs either for all subnets or no subnets.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1601)

### CODE REVIEW

File: pkg/apis/kops/validation/helpers.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import (
+	"k8s.io/kubernetes/pkg/util/validation/field"
+	"net"
+	"net/url"
+)
+
+// isSubnet checks if child is a subnet of parent
+func isSubnet(parent *net.IPNet, child *net.IPNet) bool {
+	parentOnes, parentBits := parent.Mask.Size()
```
Comment (justinsb) [inline pkg/apis/kops/validation/helpers.go]: I'll add a unit test instead

File: pkg/apis/kops/validation/legacy.go [hunk 2]
```diff
@@ -383,22 +385,16 @@ func (c *Cluster) Validate(strict bool) error {
...
-	return nil
-}
-
-func isValidAPIServersURL(s string) bool {
-	u, err := url.Parse(s)
-	if err != nil {
-		return false
+	errs := newValidateCluster(c)
+	if len(errs) != 0 {
+		return errs[0]
```
Comment (justinsb) [inline pkg/apis/kops/validation/legacy.go]: That's why it's legacy!

**Comment by justinsb** [comment]
Added test

File: pkg/apis/kops/validation/legacy.go (hunk 2 above)
Comment (chrislovecnm) [inline pkg/apis/kops/validation/legacy.go]: OMG ...

**Pull request merged**